Tools/refcounts/lint.py

```diff
@@ -0,0 +1,301 @@
+"""Lint Doc/data/refcounts.dat."""
+
+from __future__ import annotations
+
+import itertools
+import re
+import tomllib
+from argparse import ArgumentParser, RawDescriptionHelpFormatter
+from dataclasses import dataclass, field
+from enum import auto as _auto, Enum
+from pathlib import Path
+from typing import TYPE_CHECKING, LiteralString, NamedTuple
+
+if TYPE_CHECKING:
+    from collections.abc import Callable, Iterable, Mapping
+
+ROOT = Path(__file__).parent.parent.parent.resolve()
+DEFAULT_REFCOUNT_DAT_PATH: str = str(ROOT / 'Doc/data/refcounts.dat')
+DEFAULT_STABLE_ABI_TOML_PATH: str = str(ROOT / 'Misc/stable_abi.toml')
+
+C_ELLIPSIS: LiteralString = '...'
+
+MATCH_TODO: Callable[[str], re.Match | None]
+MATCH_TODO = re.compile(r'^#\s*TODO:\s*(\w+)$').match
+
+OBJECT_TYPES: frozenset[str] = frozenset()
```

Comment:
I don't think this should be a `frozenset`. All operations on it (via `|=`) are going to copy instead of updating in place. Also, you can use a class getitem:
```suggestion
OBJECT_TYPES = set[str]()
```

Reply:
I really want a frozenset here because I don't want the script mutating the variable my mistake. And I used the `|=` to indeed have a copy (and I don't really want to have `x = frozenset(x)` because of mypy).

Reply:
(And I'm not very fond of not having an explicit type annotation)

Reply:
How dare the Python script not be perfectly memory efficient! :laughing: